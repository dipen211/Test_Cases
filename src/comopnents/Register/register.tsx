import React from 'react';
export interface User {
    firstname: string;
    lastname: string;
    email: string;
    subscribed: boolean;
}
class Register extends React.Component<{}, User> {
    constructor(props: User) {
        super(props);
        this.state = {
            firstname: '',
            lastname: '',
            email: '',
            subscribed: false
        };
        this.handleChange = this.handleChange.bind(this);
        this.handleSubmit = this.handleSubmit.bind(this);
    }

    handleChange(event: any) {
        this.setState(
            {
                ...this.state,
                [event.target.name]: event.target.value,
            }
        );
    }

    handleSubmit() {
        var subscribed = this.state.subscribed ? 'Yes' : 'No';
        alert(`Firstname: ${this.state.firstname}, Lastname: ${this.state.lastname}, Email: ${this.state.email}, Subscribed: ${subscribed}`);
    }

    render() {
        return (
            <form onSubmit={this.handleSubmit}>
                <input className="form-control" type="text" name="firstname" id="firstname" value={this.state.firstname} onChange={this.handleChange} />
                <input className="form-control" type="text" name="lastname" id="lastname" value={this.state.lastname} onChange={this.handleChange} />
                <input className="form-control" type="email" name="email" id="email" value={this.state.email} onChange={this.handleChange} />
                <button type="submit" className="btn btn-primary form-control">Submit</button>
            </form>
        )
    }
}

export default Register;