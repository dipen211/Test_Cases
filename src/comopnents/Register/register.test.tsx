import React from 'react';
import Enzyme, { mount, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Register, { User } from '../../comopnents/Register/register';
import { mountToJson } from 'enzyme-to-json';

Enzyme.configure({ adapter: new Adapter() });
describe('<Register>', function () {
    let wrapper: any;
    beforeEach(() => {
        wrapper = shallow(<Register />);
    });
    it('should match the snapshot', () => {
        expect(mountToJson(wrapper)).toMatchSnapshot();
    });
    it('handle Change', () => {
        wrapper.instance().handleChange({ target: { value: 'test', name: 'email' } });
        wrapper.update();
        expect(wrapper.state('email')).toEqual('test');
    });
    it('Should capture firstname correctly onChange', () => {
        wrapper.find('input').at(0);
        wrapper.simulate('change');
        wrapper.setState({ firstname: "Dipen" })
        expect(wrapper.state('firstname')).toEqual('Dipen');
    })
    it('Should capture lastname correctly onChange', () => {
        wrapper.find('input').at(1);
        wrapper.simulate('change');
        wrapper.setState({ lastname: "Dipen" })
        expect(wrapper.state('lastname')).toEqual('Dipen');
    })
    it('Should capture email correctly onChange and change the state accordingly', () => {
        wrapper.find('input').at(2);
        wrapper.simulate('change');
        wrapper.setState({ email: "Dipen@gmail,com" })
        expect(wrapper.state('email')).toEqual('Dipen@gmail,com');
    })

    it('Should call alert() when submit button is clicked', () => {
        const state = {
            firstname: 'Dipen', lastname: 'Dipen',
            email: 'Dipen@gmail.com',
            subscribed: false
        };
        const expectedArg = "Firstname: Dipen, Lastname: Dipen, Email: Dipen@gmail.com, Subscribed: No";
        window.alert = jest.fn();
        wrapper.setState(state)

        wrapper.find('form').simulate('submit')
        expect(window.alert).toHaveBeenCalledWith(expectedArg);
    })
})