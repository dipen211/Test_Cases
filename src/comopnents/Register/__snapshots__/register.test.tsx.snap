// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Register> should match the snapshot 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    className="form-control"
    id="firstname"
    name="firstname"
    onChange={[Function]}
    type="text"
    value=""
  />
  <input
    className="form-control"
    id="lastname"
    name="lastname"
    onChange={[Function]}
    type="text"
    value=""
  />
  <input
    className="form-control"
    id="email"
    name="email"
    onChange={[Function]}
    type="email"
    value=""
  />
  <button
    className="btn btn-primary form-control"
    type="submit"
  >
    Submit
  </button>
</form>
`;
